src/App.tsx

```diff
@@ -1,29 +1,112 @@
-import { useSelector } from 'react-redux';
+/* eslint-disable max-len */
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
-import { Start } from './components/Start';
-import { Finish } from './components/Finish';
-
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoModal } from './components/TodoModal';
+import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { Filter } from './types/Filter';
 import { selectors } from './store';
+import { actions as loadingActions } from './store/loadingReducer';
+import { actions as currentTodoReducer } from './store/currentTodoReducer';
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [userId, setUserId] = useState(0);
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [filteredBy, setFilteredBy] = useState('all');
+  const [query, setQuery] = useState('');
+
+  const isLoading = useSelector(selectors.getLoading);
+  const selectedTodo = useSelector(selectors.getTodo);
```

Comment:
Create custom hook. For example useTodos and return all information your need about todos from there.
`const {isloading, selectedTodo, todo} = useTodos`